app/views/containers/steps/_image_hub_tab.html.erb

```diff
@@ -4,50 +4,59 @@
   :url => wizard_path,
   :method => :put,
   :html => {:data => {:registry => registry}} do |f| %>
+
+  <% model.errors.messages.each do |field, field_errors| %>
+    <div class="alert alert-danger">
+      <span class="pficon pficon-error-circle-o"></span>
+      <ul>
+        <% field_errors.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
   <% if registry == "registry" -%>
     <div class="input-group col-md-6">
       <%= select_registry f %>
     </div>
   <% end -%>
 
-  <div class="form-group">
-    <div class='row'>
-      <% help_type = f.object.errors[:repository_name].present? ? :help_block : :help_inline %>
-      <%= text_f(
-        f,
-        :repository_name,
-        :label => _('Search'),
-        :size => 'col-md-6',
-        :'data-url'  => auto_complete_repository_name_image_search_path(model.compute_resource_id),
-        :value       => f.object.repository_name.present? ? f.object.repository_name : '',
-        :'data-registry' => registry,
-        :'data-search' => true,
-        :focus_on_load => true,
-        :placeholder => _('Find your favorite container, e.g. centos'),
-        help_type => link_to_function(
-            icon_text('search', ''),
-            'searchRepo(this)',
-            :class => 'btn btn-default',
-            :id => "search_repository_#{registry}",
-            :'data-registry' => registry,
-            :'data-url' => search_repository_image_search_path(model.compute_resource_id)
-          ) + content_tag(:span, '', :id => 'image-confirmation').html_safe) %>
-    </div>
-    <div class='row'>
-      <%= text_f f, :tag,
-        :'data-registry' => registry,
-        :'data-tag' => true,
-        :'data-url' => auto_complete_image_tag_image_search_path(model.compute_resource_id) %>
-    </div>
+  <%= text_f(f, :repository_name,
+    :label => _('Search'),
+    :wrapper_class => (image_search_wrapper_class(model) if tab_active?(registry)),
+    :'data-url'  => auto_complete_image_search_path(model.compute_resource_id),
+    :value       => f.object.repository_name.present? ? f.object.repository_name : '',
+    :'data-registry' => registry,
+    :'data-image' => true,
+    :'data-min-length' => 1,
+    :focus_on_load => true,
+    :placeholder => _('Find your favorite container, e.g. centos'),
+    :control_group_id => "#{registry}_image_search",
+    :help_inline => spinner('', class: 'hide autocomplete-status')) %>
+
+  <%= text_f f, :tag,
+    :control_group_id => "#{registry}_tag_search",
+    :'data-registry' => registry,
+    :'data-tag' => true,
+    :'data-min-length' => 0,
+    :'data-url' => auto_complete_image_search_path(model.compute_resource_id),
+    :help_inline => spinner('', class: 'hide autocomplete-status') %>
+
+  <div class="col-md-6 col-md-offset-1">
```

Comment:
@dLobatog I prefer it aligned with the fields vs the label. It's not typical to see buttons aligned with labels in the application. Additionally -- the icons on the tabs - that's also unusual.